src/json/to_json.cr

```diff
@@ -286,3 +286,31 @@ module String::RawConverter
     json.raw(value)
   end
 end
+
+# Converter to be used with `JSON.mapping` and `YAML.mapping` to read JSON
+# values into a `Union` type with a non-standard type order.
+#
+# Default conversion prioritises component types based on an alphabetical order
+# due to type system behaviour. This may be used to to specify an alternative
+# order as required.
+#
+# ```
+# require "json"
+#
+# class Example
+#   JSON.mapping({
+#     value:           {type: Int64 | FLoat64},
+#     converted_value: {type: Int64 | FLoat64, converter: Union::OrderedConvert(Int64, Float64)},
+#   })
+# end
+#
+# example = Example.from_json(%({"value": 1, "converted_value": 1}))
+# example.value           # => "1.0"
+# example.converted_value # => "1"
+# example.to_json         # => %({"value": 1.0, "converted_value": 1})
+# ```
+module Union::OrderedConverter(*T)
+  def self.to_json(value : Union, json : JSON::Builder)
+    value.to_json(json4)
```

Comment:
json4 ? :smiley:

Reply:
As above. Evidentially this needs specs :)